### DISCUSSION TITLE
Terragrunt posting error - not sure how to solve

### DISCUSSION THREAD

**Discussion by mth11** [gruntwork-io/knowledge-base#608] [Help]
Sorry for all the newb issues, just inherited an **Azure** Terraform\Terragrunt deployment that is having issues. Not familiar with Terraform or Terragrunt and looking for some help. The environment has been operational for about 3 plus years now.

First off, can I get help here?

Second, I am just going to dive in as this is preventing an AKS update to our staging and production environments. any guidance on how to solve this issue would be appreciated.

![image](https://user-images.githubusercontent.com/50173655/206019339-8151f2f5-471f-4962-96aa-3ce2870f6f2e.png)

To follow up, I also ran the command indicated, terreaform init, and it returns:
   **Terraform initialized in an empty directory**

then running the Terragrunt plan command above returns the same message.

**Comment by gitsstewart** [comment]
I'm sorry to say that we don't currently have any modules for Azure. Terragrunt is one of our public repositories, and you can access it [here](https://github.com/gruntwork-io/terragrunt) to see how we do things with AWS. And for our product, we have [this example](https://github.com/gruntwork-io/terragrunt-infrastructure-live-example/issues/7) that relates to the initializing in an empty directory issue.

There is a _lot_ of information online about Terraform specifically. 
Here are a couple of things I got from a quick Google search of the 2 issues you mentioned (Inconsistent dependency lock file ; Terraform initialized in an empty directory):
https://stackoverflow.com/questions/70922743/inconsistent-dependency-when-i-do-terraform-apply-from-plan-out-file
https://developer.hashicorp.com/terraform/language/modules/sources

If you are interested in diving more into Terraform itself (which is what Terragrunt runs off of) I highly recommend the [Terraform: Up and Running](https://www.amazon.com/Terraform-Running-Writing-Infrastructure-Code-dp-1098116747/dp/1098116747/ref=dp_ob_title_bk) book by one of our co-founders.

**Comment by denis256** [comment]
Hi,
I got a similar error when I used the same `key` in `remote_state` between different modules
How looks configuration of `remote_state` in your case? `key` value is constant? or it is dynamically evaluated based on some variables?
